designer/server/src/common/helpers/auth/get-user-session.js

```diff
@@ -1,24 +1,69 @@
+import { token } from '@hapi/jwt'
+
 /**
- * @this {{ server: import('@hapi/hapi').Server, state?: { userSession?: { sessionId?: string } } }}
- * @returns {Promise<UserSession | undefined>}
+ * @param {Request} request
+ * @param {{ sessionId: string, user: UserCredentials }} [session] - Session cookie state
  */
-async function getUserSession() {
-  return this.state?.userSession?.sessionId
-    ? await this.server.app.cache.get(this.state.userSession.sessionId)
-    : {}
+export async function getUserSession(request, session) {
+  const { auth, server } = request
+
+  // Check for existing user
+  if (getUser(auth.credentials)) {
+    return auth.credentials
+  }
+
+  // Prefer Session ID from cookie state
+  let sessionId = session?.sessionId
+
+  // Fall back to OpenID Connect (OIDC) claim
+  if (!sessionId) {
+    const claims = getUserClaims(auth.credentials)
+
+    if (claims?.sub) {
+      sessionId = claims.sub
+    }
+  }
```

Comment:
When a user authenticates we don't know if they have a Redis session already

E.g. Maybe their token (or cookie) has expired but the Redis session hasn't

We try and grab their `sessionId` from a few places